### DISCUSSION TITLE
Get/Post requests available in current state?

### DISCUSSION THREAD

**Discussion by Dark1886** [OMOTE-Community/OMOTE-Firmware#21] [General]
Is there currently any method for creating Get/Post requests? IF not, it may be a good idea to be able to add this, It could allow some users to send at least a few simple commands if they don't have some external Homeassistant/other automation system to bridge that gap.

**Comment by CoretechR** [comment]
This could probably be done relatively easily using the HTTPClient library like this: 
https://randomnerdtutorials.com/esp32-http-get-post-arduino/
Would you like it if more examples like this are included in the main project? Or maybe some kind of documentation on how to customize the UI? I am just thinking about how I can make it easier to modify the UI until some high-level tool can help with this.

**Reply by Dark1886** [reply]
I'm not sure it would be super beneficial, I think the time is just better spent on making the project more customizable before making any sort of documentation yet. Although maybe a few QUICK comments on how to actually update the LVGL UI.

**Comment by MatthewColvin** [comment]
@Dark1886 made #22

**Comment by ramonflores08** [comment]
What I ended up doing is changing the switch to a button and then configuring it in node-red to check the state of the light and turn it on or off as needed.

this is the code how i changed it to add the power symbol.


  lv_label_set_text(menuLabel, "Bar Lights");
  lv_obj_align(menuLabel, LV_ALIGN_TOP_LEFT, 22, 3);
  lv_obj_t* lightToggleA = lv_btn_create(menuBox);
  lv_obj_set_size(lightToggleA, 40, 30);
  buttonLabel = lv_label_create(lightToggleA);
  lv_label_set_text(buttonLabel, LV_SYMBOL_POWER);
  lv_obj_align(lightToggleA, LV_ALIGN_TOP_RIGHT, 0, 0);
  lv_obj_set_style_bg_color(lightToggleA, lv_color_lighten(color_primary, 50), LV_PART_MAIN);
  lv_obj_set_style_bg_color(lightToggleA, color_primary, LV_PART_INDICATOR);
  lv_obj_add_event_cb(lightToggleA, smartHomeToggle_event_cb, LV_EVENT_CLICKED, (void*)1);

![20230821_150501](https://github.com/CoretechR/OMOTE/assets/7199615/8203903a-4083-469c-8e28-3e893f72ff21)

**Comment by tnph** [comment]
Hi @ramonflores08 - this is similar to what I'm trying to achieve, ie. a toggle instead of a switch, being a non-coder I find the code quite un-intuitive!

In my case I was publishing an MQTT message to turn power on like "omote/power_on" and one to turn off "omote/power_off" but the problem is the remote doesn't know the current state and defaults to off so they get out of sync.  I'm using Home Assistant to trigger on/off actions when it sees the relevant MQTT messages, but I will copy your power button idea and simply use the toggle action in HA. Thanks!  

I'm also keen to get the Battery charge % display working - how did you do that?  I tried un-commenting out this line of code

lv_label_set_text_fmt(objBattPercentage, "%d%%", battery_percentage);

..but it didn't really work - just flashed up random percentages
